app/main.py

```diff
@@ -1,25 +1,73 @@
+from __future__ import annotations
+
 from abc import ABC
+from typing import Callable
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: SlideLimitationValidator, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(
+            self,
+            instance: SlideLimitationValidator,
+            owner: SlideLimitationValidator,
+    ) -> int:
```

Comment:
`owner`  redundant here
```suggestion
    def __get__(self, instance: SlideLimitationValidator) -> int:
```